Do not restrict the files name to letters and numbers in fluentd

aiwantaozi opened the pull request:
Problem:
now filename only allow numbers and letter

Solution:
change regex expression to allow all charaters

Issue:
https://github.com/rancher/rancher/issues/17022

aiwantaozi committed:
copy from 0.1.1

gitlawr commented on charts/rancher-logging/0.1.2/charts/fluentd/values.yaml:
```diff
@@ -143,22 +143,22 @@ secrets:
-          volume_name ${component = tag.match(Regexp.compile('tmp-cluster-custom\.var\.lib\.rancher\.log-volumes\.(?<pod_id>[^_]+)_(?<volume_name>[^_]+)\.(?<format>[a-z0-9]+)\.(?<cluster_id>[^_]+)_(?<cluster_name>[^_]+)_(?<namespace>[^_]+)_(?<project_id>[^_]+)_(?<project_name>[^_]+)_(?<workload_name>[^_]+)_(?<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])+)_(?<container_name>[^_^.]+)\.(?<filename>[a-z0-9A ...
-          filename ${component = tag.match(Regexp.compile('tmp-cluster-custom\.var\.lib\.rancher\.log-volumes\.(?<pod_id>[^_]+)_(?<volume_name>[^_]+)\.(?<format>[a-z0-9]+)\.(?<cluster_id>[^_]+)_(?<cluster_name>[^_]+)_(?<namespace>[^_]+)_(?<project_id>[^_]+)_(?<project_name>[^_]+)_(?<workload_name>[^_]+)_(?<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])+)_(?<container_name>[^_^.]+)\.(?<filename>[a-z0-9A-Z] ...
-          format ${component = tag.match(Regexp.compile('tmp-cluster-custom\.var\.lib\.rancher\.log-volumes\.(?<pod_id>[^_]+)_(?<volume_name>[^_]+)\.(?<format>[a-z0-9]+)\.(?<cluster_id>[^_]+)_(?<cluster_name>[^_]+)_(?<namespace>[^_]+)_(?<project_id>[^_]+)_(?<project_name>[^_]+)_(?<workload_name>[^_]+)_(?<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])+)_(?<container_name>[^_^.]+)\.(?<filename>[a-z0-9A-Z]+\ ...
+          cluster_id ${component = tag.match(Regexp.compile('tmp-cluster-custom\.var\.lib\.rancher\.log-volumes\.(?<pod_id>[^_]+)_(?<volume_name>[^_]+)\.(?<format>[a-z0-9]+)\.(?<cluster_id>[^_]+)_(?<cluster_name>[^_]+)_(?<namespace>[^_]+)_(?<project_id>[^_]+)_(?<project_name>[^_]+)_(?<workload_name>[^_]+)_(?<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])+)_(?<container_name>[^_^.]+)\.(?<filename>[^^]+)')) ...
```
Are we excluding the `^` character here(`[^^]+`)? I think it is valid in a filename.

aiwantaozi replied:
Yes, because caret is invalid in some filesystem, like Windows using the FAT file system

gitlawr replied:
I see

gitlawr replied:
If it is invalid, we don't need to care about it so as to match the filenames. Doing that will break when the character is valid.

gitlawr commented:
LGTM

aiwantaozi committed:
Do not restrict the files name to letters and numbers in fluentd

Problem:
now filename only allow numbers and letter

Solution:
change regex expression to allow all charaters

Issue:
https://github.com/rancher/rancher/issues/17022

gitlawr commented:
LGTM

cjellick commented:
lgtm
